### PR TITLE
enhance/mobile-detailed-page

### PR DESCRIPTION
**Pull request by haritonasty**
**Summary**
New flow on mobile detailed page: 
- fullscreen
- metrics selector
- swipes

### CODE REVIEW

File: src/components/MobileMetricCard/MobileMetricCard.js [hunk 1]
```diff
@@ -1,73 +1,159 @@
...
 import Label from '@santiment-network/ui/Label'
+import Dialog from '@santiment-network/ui/Dialog'
 import { formatTooltipValue } from '../../ducks/SANCharts/CustomTooltip'
-import { Metrics, compatabilityMap } from '../../ducks/SANCharts/data'
 import PercentChanges from '../PercentChanges'
+import { DAY, getTimeIntervalFromToday } from '../../utils/dates'
+import { calcPercentageChange } from '../../utils/utils'
+import { makeRequestedData } from '../../pages/Detailed/mobile/utils'
 import { METRIC_ANOMALIE_QUERY } from '../../ducks/GetTimeSeries/queries/metric_anomaly_query'
+import Loader from '@santiment-network/ui/Loader/Loader'
```
Comment (Partysun) [inline src/components/MobileMetricCard/MobileMetricCard.js]: Move such import above internal imports @haritonasty

File: src/components/MobileMetricCard/MobileMetricCard.js [hunk 2]
```diff
@@ -1,73 +1,159 @@
...
+                  <>
+                    <h4 className={styles.value}>{value}</h4>
+                    <PercentChanges changes={diff} />
+                    <Label accent='casper' className={styles.period}>
+                      , {period || '24h'}
+                    </Label>
+                  </>
+                )}
+                {isLoading && !value && <Loader className={styles.loader} />}
+                {false && !hasPremium && !isLoading && !value && (
```
Comment (Partysun) [inline src/components/MobileMetricCard/MobileMetricCard.js]: False? @haritonasty

File: src/ducks/SANCharts/ChartMetricSelector.js [hunk 3]
```diff
@@ -161,95 +125,177 @@ const ChartMetricSelector = ({
...
+          />
+        </div>
+      )}
       <Panel.Content className={cx(styles.wrapper, className)}>
-        {loading && (
-          <div className={styles.loader}>
-            <Loader />
+        {loading && <Loader className={styles.loader} />}
+        {isMobile ? (
+          <div className={cx(styles.column, styles.categories)}>
```
Comment (Partysun) [inline src/ducks/SANCharts/ChartMetricSelector.js]: @haritonasty Put Mobile render outside

File: src/pages/Detailed/mobile/MobilePopularMetrics.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  const hideTooltip = () => {
+    localStorage.setItem(LS_LABEL, '+')
+    setIsShow(false)
+  }
+
+  useEffect(() => {
+    if (!wasShown) {
+      setIsShow(true)
+    }
```
Comment (DmitriVanGuard) [inline src/pages/Detailed/mobile/MobilePopularMetrics.js]: `localStorage.getItem(LS_LABEL)` could be called only on component mount (currently on every render)

File: src/pages/Detailed/mobile/MobilePopularMetrics.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  }, [])
+
+  return (
+    <>
+      <h3 className={styles.heading}>Popular metrics</h3>
+      {isShow && !wasShown && (
```
Comment (DmitriVanGuard) [inline src/pages/Detailed/mobile/MobilePopularMetrics.js]: `!wasShown` is redundant

File: src/ducks/SANCharts/IntervalSelector.js [hunk 6]
```diff
@@ -8,14 +8,14 @@ export const INTERVAL_ALIAS = {
...
-export const getNewInterval = (from, to, lastInterval) => {
+export const getNewInterval = (from, to, lastInterval, options = {}) => {
   const intervals = getAvailableIntervals(from, to)
 
   if (intervals.includes(lastInterval)) {
     return lastInterval
   }
 
-  return intervals[0]
+  return options.isMobile ? intervals[1] : intervals[0]
```
Comment (GarageInc) [inline src/ducks/SANCharts/IntervalSelector.js]: Strange indexes -_-

File: src/components/MobileMetricCard/SwipeableCard.js [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+      return
+    }
+
+    setOffset(offset)
+    setSide(offset > 0 ? 'left' : 'right')
+
+    const ts = Date.now()
+    const speed = Math.abs(x - prevX) / (ts - prevTs)
+
+    if ((speed > 4 && ts !== prevTs) || (Math.abs(offset) * 100) / width > 85) {
```
Comment (DmitriVanGuard) [inline src/components/MobileMetricCard/SwipeableCard.js]: I guess it's better to move these checks to the function. Right now, it's hard to understand its purpose on a first glance

File: src/components/MobileMetricCard/SwipeableCard.js [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+      setCurrentGesture(null)
+      return
+    }
+
+    setCurrentGesture({ ...currentGesture, prevX: x, prevTs: ts })
+
+    if (dx < -1.5 * BUTTON_WIDTH) {
+      setStartPosition(-BUTTON_WIDTH)
+    } else if (dx > 1.5 * BUTTON_WIDTH) {
+      setStartPosition(BUTTON_WIDTH)
```
Comment (DmitriVanGuard) [inline src/components/MobileMetricCard/SwipeableCard.js]: `1.5 * BUTTON_WIDTH` is a constant expression and could be moved out. The only variable name I came up with is `BUTTON_BORDER_AREA`, but it's not so good :(.

I'm not sure about following suggestion, but here it is:
```js
if( Math.abs(dx) > BUTTON_BORDER_AREA ){
  setStartPosition(dx > 0 ? BUTTON_WIDTH : -BUTTON_WIDTH)
}
```

File: src/utils/utils.js [hunk 9]
```diff
@@ -240,7 +240,8 @@ const getTimeFromFromString = (time = '1y') => {
   return time
 }
 
-const capitalizeStr = string => string.charAt(0).toUpperCase() + string.slice(1)
```
Comment (GarageInc) [inline src/utils/utils.js]: Can be here 'null' or empty string?

**Pull request merged**